skfda/exploratory/visualization/representation.py

```diff
@@ -119,79 +120,101 @@ def _get_color_info(
+            if fig and ax are None.
+        n_cols(int, optional): designates the number of columns of the
+            figure to plot the different dimensions of the image. Only
+            specified if fig and ax are None.
+    Attributes:
+        gradient_list: normalization of the values from gradient color_list
+            that will be used to determine the intensity of the color
+            each function will have.
     """
 
     def __init__(
         self,
         fdata: FData,
-        gradient_values: Optional[Sequence[float]] = None,
+        gradient_color_list: Union[Sequence[float], None] = None,
         max_grad: Optional[float] = None,
         min_grad: Optional[float] = None,
+        chart: Union[Figure, Axes, None] = None,
+        *,
+        fig: Optional[Figure] = None,
+        axes: Optional[Axes] = None,
+        n_rows: Optional[int] = None,
+        n_cols: Optional[int] = None,
     ) -> None:
+        BasePlot.__init__(self)
         self.fdata = fdata
-        self.gradient_values = gradient_values
-        if self.gradient_values is not None:
-            if len(self.gradient_values) != fdata.n_samples:
+        self.gradient_color_list = gradient_color_list
+        if self.gradient_color_list is not None:
+            if len(self.gradient_color_list) != fdata.n_samples:
                 raise ValueError(
-                    "The length of the gradient color"
-                    "list should be the same as the number"
+                    "The length of the gradient color",
+                    "list should be the same as the number",
                     "of samples in fdata",
                 )
 
             if min_grad is None:
-                self.min_grad = min(self.gradient_values)
+                self.min_grad = min(self.gradient_color_list)
             else:
                 self.min_grad = min_grad
 
             if max_grad is None:
-                self.max_grad = max(self.gradient_values)
+                self.max_grad = max(self.gradient_color_list)
             else:
                 self.max_grad = max_grad
 
             aux_list = [
                 grad_color - self.min_grad
-                for grad_color in self.gradient_values
+                for grad_color in self.gradient_color_list
             ]
 
-            self.gradient_list = (
+            self.gradient_list: Sequence[float] = (
                 [
                     aux / (self.max_grad - self.min_grad)
                     for aux in aux_list
                 ]
             )
         else:
             self.gradient_list = []
+        self._set_figure_and_axes(chart, fig, axes, n_rows, n_cols)
 
     def plot(
         self,
-        chart: Union[Figure, Axes, None] = None,
         *,
-        fig: Optional[Figure] = None,
-        ax: Optional[Axes] = None,
-        n_rows: Optional[int] = None,
-        n_cols: Optional[int] = None,
         n_points: Union[int, Tuple[int, int], None] = None,
```

Comment:
What criterion do we use for arguments in `__init__` vs `plot`?

Reply:
I left colors and groups here, besides I also left the domain range or n_points in case someone wanted to plot the same and modify that parameters, but perhaps it is a possibility to move domain_range and n_points to the init as they aren't color or group settings

Reply:
Unless there is a very good reason not to, I would move all of them to the `__init__` also, and allow the modification of them editing the attributes of the object.

Reply:
ok, just moved them